lib/get-config.js

```diff
@@ -303,7 +305,25 @@ function validateRules(config, options) {
   }
 }
 
-module.exports = function(options) {
+/**
+ * Returns the config in conjunction with overrides configuration.
+ * @param {*} config
+ * @param {*} filePath
+ */
+function getConfigForFile(config, filePath) {
+  let configuredRules = config.rules;
+  let overrides = config.overrides;
+  if (filePath && overrides) {
+    let overridesObj =
+      overrides.find(override => override.files && micromatch.isMatch(filePath, override.files)) ||
+      {};
+    // If there is an override present, then the overridden ruleset takes precedence.
+    config.rules = Object.assign({}, configuredRules, overridesObj.rules);
```

Comment:
I worry about mutating here, can we return a new object?

I'm thinking basically:

```suggestion
    config = Object.assign({}, config, {
      rules: Object.assign({}, configuredRules, overridesObj.rules),
    });
```